### PR TITLE
Init and switch project flow

### PR DESCRIPTION
**Pull request by pbaranski9411**

### CODE REVIEW

File: backend/aiconsole/api/endpoints/projects/choose.py [hunk 1]
```diff
@@ -14,62 +14,36 @@
...
 
-    is_project = directory.joinpath("materials").is_dir() or directory.joinpath("agents").is_dir()
+@router.get("/initial_directory")
+async def initial_directory(project_directory: ProjectDirectory = Depends(project_directory)):
+    initial_directory = await project_directory.get_initial_directory()
+    return {"directory": None if initial_directory is None else str(initial_directory)}
 
-    return {"is_project": is_project, "path": str(directory)}
 
+@router.post("/in_directory")
```
Comment (pbaranski9411) [inline backend/aiconsole/api/endpoints/projects/choose.py:36]: @cielecki I think it should be get request, but as always there is a problem with passing path through url.
Do we have some ID -> path mapping for projects or something which is similar to having something in db? 🤔

Comment (cielecki) [reply backend/aiconsole/api/endpoints/projects/choose.py:36]: We don't have ids. Why does lack of ids is related to this call?

This call should take a path and return if a given path contains an existing project. the name of the endpoint might need improvement.

File: backend/aiconsole/api/endpoints/projects/choose.py [hunk 2]
```diff
@@ -14,62 +14,36 @@
...
-        # Show a system select directory dialog
-        directory = await _ask_directory()
+class ProjectDirectoryParams(BaseModel):
+    directory: str
 
-    if not directory:
-        return {"is_project": False, "path": None}
 
-    is_project = directory.joinpath("materials").is_dir() or directory.joinpath("agents").is_dir()
+@router.get("/initial_directory")
```
Comment (cielecki) [inline backend/aiconsole/api/endpoints/projects/choose.py:30]: What does this end point do?

Comment (pbaranski9411) [reply backend/aiconsole/api/endpoints/projects/choose.py:30]: This is something that was previously done by "in_project" endpoint. This is running Tk stuff to open initial directory, but now I'm thinking if name like: "choose_directory" with POST method would be better for that action. What do you think? Or you think it's not needed at all? 🤔

File: backend/aiconsole/api/endpoints/projects/choose.py (hunk 1 above)
Comment (pbaranski9411) [reply backend/aiconsole/api/endpoints/projects/choose.py:36]: Yeah, it's ok. Just asking if I understand it properly :)

What do you think about endpoint: GET `/projects/is_in_directory?directory=...`?
f

Comment (cielecki) [reply backend/aiconsole/api/endpoints/projects/choose.py:36]: Oh yeah I always forget that GETs can not have JSON body, this was the reason it was a POST in the first place

File: backend/aiconsole/api/endpoints/projects/choose.py (hunk 2 above)
Comment (cielecki) [reply backend/aiconsole/api/endpoints/projects/choose.py:30]: choose_directory is great

File: backend/aiconsole/api/endpoints/projects/choose.py (hunk 1 above)
Comment (pbaranski9411) [reply backend/aiconsole/api/endpoints/projects/choose.py]: Done

File: backend/aiconsole/api/endpoints/projects/choose.py (hunk 2 above)
Comment (pbaranski9411) [reply backend/aiconsole/api/endpoints/projects/choose.py]: Done

**Review by cielecki — APPROVED** [review]

**Pull request merged**